ironfish/src/blockchain/database/assets.ts

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import type { IDatabaseEncoding } from '../../storage/database/types'
+import { PUBLIC_ADDRESS_LENGTH } from '@ironfish/rust-nodejs'
+import bufio from 'bufio'
+import { BigIntUtils } from '../../utils'
+
+export interface AssetsValue {
+  createdTransactionHash: Buffer
+  metadata: string
+  name: string
+  nonce: number
+  owner: string
+  supply: bigint
+}
+
+export class AssetsValueEncoding implements IDatabaseEncoding<AssetsValue> {
+  serialize(value: AssetsValue): Buffer {
+    const bw = bufio.write(this.getSize(value))
+    bw.writeHash(value.createdTransactionHash)
+    bw.writeVarString(value.metadata, 'utf8')
+    bw.writeVarString(value.name, 'utf8')
```

Comment:
how is name a var string? Is this a var string in our rust side and not a fixed length buffer?

Reply:
It gets converted to a fixed length buffer in the Rust side. It's functionally identical to the way `memo` works on `Note`

Reply:
Ok, what do you think about down the line making a pass and converting all these strings to buffers? They're fixed size  and don't need to contain human readable data?